install/kubernetes/helm/istio/charts/mixer/templates/config.yaml

```diff
@@ -467,9 +465,8 @@ spec:
       destination_service_namespace: destination.service.namespace | "unknown"
       request_protocol: api.protocol | context.protocol | "unknown"
       response_code: response.code | 200
+      grpc_response_status: response.grpc_status | 0
```

Comment:
is the status an int or a string? I think it's a string (by mistake)

Reply:
apparently it is a string (based on the test failures).  Is "0" the right default, or "unknown" ?

Reply:
it's set if grpc status header is present, and not set at all otherwise (see `
`src/envoy/http/mixer/report_data.h`)

Reply:
right, but do we want the unset to be represented in metrics as "0" (like we default to "200" for http) or "unknown" ?

Reply:
I think empty is better. It shouldn't really be unknown for grpc ever.

Reply:
So just to confirm, the decision to have separate response_code and grpc_response_status fields is because even if request_protocol=grpc, response.code is still set?

Also, I think 0 is good for unset, just like the 200.  I like a default being the same type as an actual value.

Reply:
@douglas-reid I know the ship has sailed but perhaps the new field should have been "request_protocol_status" or something like that.  And contained the response status for the specified request_protocol.  Basically replacing request_code, which has now been relegated to effectively being "http_code".  So existing consumer code could still assume http codes in request_code and write new code using the new field, which could support the two existing protocols and if necessary, additional protocols somewhere down the line.  And "response_code" could have been deprecated.

Reply:
@jshaughn perhaps -- that's closer to my original desire. I think we can rethink that for 1.6/1.7 when maybe we do an overhaul of the metrics (and they are controlled all via v2). however, at this point, i'd like to get the approved design implemented and into istio/istio first.

Reply:
@douglas-reid yup, totally agree.  And I've been coding kiali the past days in anticipation of the new grpc_request_status field, and to be back compatible.